### PR TITLE
Pal 197 add eureka support to services

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

**Review by dev-958 — DISMISSED** [review]

**Review by m09526 — CHANGES_REQUESTED** [review]
Looks good. With regard to the version numbers in the POM files, please could these be extracted as properties into the main repo POM along with the other version numbers that are there?

File: palisade-service/mvn_dependency_tree.txt [hunk 1]
```diff
@@ -220,12 +242,18 @@ net.minidev/accessors-smart/version = 1.2
...
 net.minidev/json-smart/version = 2.3
 
+org.antlr/antlr-runtime/version = 3.4
+
+org.antlr/stringtemplate/version = 3.2.1
+
 org.apache.avro/avro/version = 1.8.2
 
-org.apache.commons/commons-compress/version = 1.18
+org.apache.commons/commons-compress/version = 1.8.1
```
Comment (m09526) [inline palisade-service/mvn_dependency_tree.txt:251]: Why has this been downgraded?

File: policy-service/mvn_dependency_tree.txt [hunk 2]
```diff
@@ -42,9 +46,9 @@ com.google.code.gson/gson/version = 2.8.5
 
 com.google.errorprone/error_prone_annotations/version = 2.2.0
 
-com.google.guava/guava/version = 26.0-android
+com.google.guava/guava/version = 16.0
```
Comment (m09526) [inline policy-service/mvn_dependency_tree.txt:49]: Significant downgrade?

File: policy-service/mvn_dependency_tree.txt [hunk 3]
```diff
@@ -116,13 +130,13 @@ io.etcd/jetcd-launcher/version = 0.3.0
 
 io.etcd/jetcd-resolver/version = 0.3.0
 
-io.fabric8/kubernetes-client/version = 4.3.0
+io.fabric8/kubernetes-client/version = 4.2.0
```
Comment (m09526) [inline policy-service/mvn_dependency_tree.txt:133]: In the other repo we are using 4.3, here we are using 4.2?

File: policy-service/mvn_dependency_tree.txt [hunk 4]
```diff
@@ -220,12 +236,18 @@ net.minidev/accessors-smart/version = 1.2
...
 net.minidev/json-smart/version = 2.3
 
+org.antlr/antlr-runtime/version = 3.4
+
+org.antlr/stringtemplate/version = 3.2.1
+
 org.apache.avro/avro/version = 1.8.2
 
-org.apache.commons/commons-compress/version = 1.18
+org.apache.commons/commons-compress/version = 1.8.1
```
Comment (m09526) [inline policy-service/mvn_dependency_tree.txt:245]: Downgraded is before?

File: resource-service/mvn_dependency_tree.txt [hunk 5]
```diff
@@ -56,11 +58,11 @@ com.google.errorprone/error_prone_annotations/version = 2.2.0
 
 com.google.guava/guava/version = 11.0.2
 
-com.google.inject/guice/version = 4.0
+com.google.inject/guice/version = 4.1.0
 
 com.google.inject.extensions/guice-servlet/version = 4.0
 
-com.google.protobuf/protobuf-java/version = 3.5.1
+com.google.protobuf/protobuf-java/version = 2.5.0
```
Comment (m09526) [inline resource-service/mvn_dependency_tree.txt:65]: Why downgrade?

File: resource-service/mvn_dependency_tree.txt [hunk 6]
```diff
@@ -136,9 +130,9 @@ commons-configuration/commons-configuration/version = 1.8
 
 commons-daemon/commons-daemon/version = 1.0.13
 
-commons-fileupload/commons-fileupload/version = 1.4
+commons-io/commons-io/version = 2.5
```
Comment (m09526) [inline resource-service/mvn_dependency_tree.txt:133]: Downgrade?

File: resource-service/mvn_dependency_tree.txt [hunk 7]
```diff
@@ -160,21 +154,13 @@ io.etcd/jetcd-core/version = 0.3.0
...
-io.github.openfeign/feign-core/version = 10.2.3
-
-io.github.openfeign/feign-hystrix/version = 10.2.3
+io.fabric8/kubernetes-client/version = 4.3.0
 
-io.github.openfeign/feign-slf4j/version = 10.2.3
+io.fabric8/kubernetes-model/version = 4.2.0
 
-io.github.openfeign.form/feign-form/version = 3.8.0
+io.fabric8/kubernetes-model-common/version = 4.2.0
```
Comment (m09526) [inline resource-service/mvn_dependency_tree.txt]: Downgraded?

File: resource-service/mvn_dependency_tree.txt [hunk 8]
```diff
@@ -504,39 +488,25 @@ org.springframework.boot/spring-boot-test-autoconfigure/version = 2.1.8.RELEASE
...
 
 org.springframework.security/spring-security-crypto/version = 5.1.6.RELEASE
 
-org.springframework.security/spring-security-rsa/version = 1.0.7.RELEASE
-
-org.tukaani/xz/version = 1.5
+org.tukaani/xz/version = 1.0
 
-org.xerial.snappy/snappy-java/version = 1.1.1.3
+org.xerial.snappy/snappy-java/version = 1.0.5
```
Comment (m09526) [inline resource-service/mvn_dependency_tree.txt:499]: Downgrade?

File: resource-service/pom.xml [hunk 9]
```diff
@@ -20,70 +20,79 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.fabric8</groupId>
+            <artifactId>kubernetes-client</artifactId>
+            <version>4.3.0</version>
+        </dependency>
+        <dependency>
+            <groupId>io.fabric8</groupId>
+            <artifactId>kubernetes-model</artifactId>
+            <version>4.2.0</version>
```
Comment (m09526) [inline resource-service/pom.xml]: These two dependencies have a version mismatch between 4.3 and 4.2

**Review by m09526 — CHANGES_REQUESTED** [review]
All code looks good but numerous libraries have been downgraded from what we had before. Please could you check to ensure common versions of things across the modules and check the version numbers where things have been downgraded? Were the downgrades a result of the requirement that all versions are at least a week old?

File: resource-service/pom.xml (hunk 9 above)
Comment (dev930018) [reply resource-service/pom.xml]: Resolved

**Review by m09526 — DISMISSED** [review]

**Review by m09526 — APPROVED** [review]

**Pull request merged**